components/eam/src/physics/cam/zm_conv.F90

```diff
@@ -1720,417 +1456,97 @@ subroutine momtran(lchnk, ncol, &
             end do
          end do
 
-  ! dcont for bottom layer
-          !
+         
 #ifdef CPRCRAY
-          !DIR$ NOINTERCHANGE
+!DIR$ NOINTERCHANGE
 #endif
-          do k = kbm,pver
-             km1 = max(1,k-1)
-             do i = il1g,il2g
-                if (k == mx(i)) then
-
-                   ! version 1
-                   dcondt(i,k) = (1._r8/dp(i,k))*   &  
-                        (-mu(i,k)*(conu(i,k)-chat(i,k)) &
-                        -md(i,k)*(cond(i,k)-chat(i,k)) &
-                        )
-                end if
-             end do
-          end do
+         ! dcont for bottom layer
+         do k = kbm,pver
+            km1 = max(1,k-1)
+            do i = il1g,il2g
+               if (k == mx(i)) then
+                  dcondt(i,k) = (1._r8/dp(i,k))*   &  
+                       (-mu(i,k)*(conu(i,k)-chat(i,k)) &
+                       -md(i,k)*(cond(i,k)-chat(i,k))  )
+               end if
+            end do
+         end do
 
-! Initialize to zero everywhere, then scatter tendency back to full array
+         ! Initialize to zero everywhere, then scatter tendency back to full array
          dqdt(:,:,m) = 0._r8
 
          do k = 1,pver
             do i = il1g,il2g
                ii = ideep(i)
                dqdt(ii,k,m) = dcondt(i,k)
-    ! Output apparent force on the mean flow from pressure gradient
+               ! Output apparent force on the mean flow from pressure gradient
                pguall(ii,k,m) = -pgu(i,k)
                pgdall(ii,k,m) = -pgd(i,k)
                icwu(ii,k,m)   =  conu(i,k)
                icwd(ii,k,m)   =  cond(i,k)
             end do
          end do
 
-          ! Calculate momentum flux in units of mb*m/s2 
```

Comment:
Note that the reformatted codes here on to the end of the subroutine momtran are all good, though they appear scattering between various segments of the unwanted (thus deleted) subroutine 'buoyan'.